test/es-module/test-typescript.mjs

```diff
@@ -301,26 +315,66 @@ test('execute a TypeScript file with CommonJS syntax requiring .mts with require
 });
 
 test('execute a TypeScript file with CommonJS syntax requiring .mts with require-module with default-type commonjs',
-     async () => {
-       const result = await spawnPromisified(process.execPath, [
-         '--experimental-strip-types',
-         '--experimental-default-type=commonjs',
-         '--no-warnings',
-         fixtures.path('typescript/ts/test-require-cts.ts'),
-       ]);
-
-       strictEqual(result.stderr, '');
-       match(result.stdout, /Hello, TypeScript!/);
-       strictEqual(result.code, 0);
-     });
-
-test('execute a JavaScript file importing a cjs TypeScript file', async () => {
+  async () => {
   const result = await spawnPromisified(process.execPath, [
     '--experimental-strip-types',
+    '--experimental-default-type=commonjs',
     '--no-warnings',
-    fixtures.path('typescript/ts/issue-54457.mjs'),
+    fixtures.path('typescript/ts/test-require-cts.ts'),
   ]);
+
   strictEqual(result.stderr, '');
   match(result.stdout, /Hello, TypeScript!/);
   strictEqual(result.code, 0);
 });
+
+test('execute a TypeScript file with Partial Union Type', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-strip-types',
+    '--no-warnings',
+    fixtures.path('typescript/ts/test-partial-union-type.ts'),
+  ]);
+
+  strictEqual(result.stderr, '');
+  match(result.stdout, /Hello, TypeScript!/);
+  strictEqual(result.code, 0);
+});
+
+test('execute a TypeScript file with Extract Union Type', async () => {
```

Comment:
can you move the union test into one?

Reply:
you say create a separate test file to test the union?